From c2c9b7a9c0d44f2e440d79a10af6e4c747e7daab Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Mon, 8 Sep 2014 16:14:10 -0400
Subject: [ethernet] be2net: fix error status for FW-download

Message-id: <1410192881-30702-48-git-send-email-ivecera@redhat.com>
Patchwork-id: 92423
O-Subject: [RHEL7.1 PATCH 47/78] be2net: fix error status for FW-download
Bugzilla: 1088575
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>

BZ: #1088575

Upstream commit(s):
commit 56ace3a0d0209ea9f1806abb9c1046fdb89e0030
Author: Kalesh AP <kalesh.purayil@emulex.com>
Date:   Thu Jul 17 16:20:20 2014 +0530

    be2net: fix error status for FW-download

    For FW download ethtool cmd, if the user provides an FW-image incompatible
    with the chip, return -EINVAL instead of -1.

    Signed-off-by: Kalesh AP <kalesh.purayil@emulex.com>
    Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/emulex/benet/be_main.c b/drivers/net/ethernet/emulex/benet/be_main.c
index 387220b..e0287b5 100644
--- a/drivers/net/ethernet/emulex/benet/be_main.c
+++ b/drivers/net/ethernet/emulex/benet/be_main.c
@@ -3952,7 +3952,7 @@ static int be_flash_skyhawk(struct be_adapter *adapter,
  fsec = get_fsec_info(adapter, filehdr_size + img_hdrs_size, fw);
  if (!fsec) {
   dev_err(dev, "Invalid Cookie. FW image may be corrupted\n");
-  return -1;
+  return -EINVAL;
  }
 
  for (i = 0; i < le32_to_cpu(fsec->fsec_hdr.num_images); i++) {
@@ -4183,7 +4183,7 @@ static int be_fw_download(struct be_adapter *adapter, const struct firmware* fw)
              &flash_cmd,
              num_imgs);
     else {
-     status = -1;
+     status = -EINVAL;
      dev_err(&adapter->pdev->dev,
       "Can't load BE3 UFI on BE3R\n");
     }
@@ -4194,7 +4194,7 @@ static int be_fw_download(struct be_adapter *adapter, const struct firmware* fw)
  if (ufi_type == UFI_TYPE2)
   status = be_flash_BEx(adapter, fw, &flash_cmd, 0);
  else if (ufi_type == -1)
-  status = -1;
+  status = -EINVAL;
 
  dma_free_coherent(&adapter->pdev->dev, flash_cmd.size, flash_cmd.va,
      flash_cmd.dma);
-- 
1.7.1